src/language_server.ts

```diff
@@ -419,4 +480,14 @@ export class LanguageServer {
 			}
 		});
 	}
+
+	cleanOutputChannel() {
```

Comment:
> If I understand correctly this will remove the logs of each LS whenever we switch workspace?

I'm not 100% what you mean exactly by "switching workspace", but here's how it works at the moment:

We only remove the logs of the ls whose folders have been removed from the workspace. e.g. you're working on a workspace with 3 folders: a project and 2 modules. Assuming they are all distinct folders (not parent of each other), then you would have 3 language servers, one per folder with the name of the folder as log title.

The cleanup would only happen if you removed one folder from the workspace (e.g. you remove the project folder, then you would only have left the logs for the 2 modules)

Reply:
ok that's nice! :)